File: posts/code/julia-no-susume-20210115/print-debug.jl
macro showln(ex)
    ln = __source__.line
    quote
        print('L', $ln, ": ")
        @show $ex
    end |> esc
end

function foo(n)
    a, ret = 0, 0
    for i in 1:n
        @showln a += i
        for j in 1:a
            @showln a += j
        end
        @showln ret += a
    end

    return ret
end

foo(parse(Int, first(ARGS)))

File: posts/code/julia-no-susume-20210115/print-debug.py
import sys

def foo(n):
    a, ret = 0, 0
    for i in range(1,n+1):
        a += i
        print(f"L6: a = {a}")
        for j in range(1,a+1):
            a += j
            print(f"L9: a = {a}")

        ret += a
        print(f"L12: ret = {ret}")

    return ret


foo(int(sys.argv[1]))
